Build failure: _2ship2harkinian

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *_2ship2harkinian*

### Build log

```
$ NIXPKGS_ALLOW_UNFREE=1 nix-build -A _2ship2harkinian

...

-- Linking against dependent libraries dynamically
-- Configuring incomplete, errors occurred!
error: builder for '/nix/store/ik9djj40i17illwidmyhrspk9c7bv48h-stormlib-9.25.drv' failed with exit code 1;
       last 10 log lines:
       > -- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
       > CMake Error at /nix/store/sa1yjb7xxrzcbd1bysj3r561vidfxxvb-cmake-3.29.6/share/cmake-3.29/Modules/FindPkgConfig.cmake:696 (message):
       >   pkg-config tool not found
       > Call Stack (most recent call first):
       >   /nix/store/sa1yjb7xxrzcbd1bysj3r561vidfxxvb-cmake-3.29.6/share/cmake-3.29/Modules/FindPkgConfig.cmake:862 (_pkg_check_modules_internal)
       >   CMakeLists.txt:315 (pkg_check_modules)
       >
       > 
       > -- Linking against dependent libraries dynamically
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/ik9djj40i17illwidmyhrspk9c7bv48h-stormlib-9.25.drv'.
error: 1 dependencies of derivation '/nix/store/qhf82ypbm5viy3qr3avkw0qlhf5dh41j-2ship2harkinian-1.0.2.drv' failed to build
```

### Additional context

I'm still learning how the build system works but I was able to fix this by changing `buildInputs` to `nativeBuildInputs` in both places on this line:
https://github.com/NixOS/nixpkgs/blob/d4a7a4d0e066278bfb0d77bd2a7adde1c0ec9e3d/pkgs/by-name/_2/_2ship2harkinian/package.nix#L80

I can open a PR with these changes if that's the correct fix.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@qubitnano

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.4-xanmod1, NixOS, 24.11 (Vicuna), 24.11.20240814.c3aa7b8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos-24.05, nixos-unstable"`
 - nixpkgs: `/nix/store/sfycwi72zfjsspidinx56ajaiffpyh17-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/_2/_2ship2harkinian/package.nix
+++ b/pkgs/by-name/_2/_2ship2harkinian/package.nix
@@ -77,7 +77,7 @@ let
       rev = "v${version}";
       hash = "sha256-HTi2FKzKCbRaP13XERUmHkJgw8IfKaRJvsK3+YxFFdc=";
     };
-    buildInputs = prev.buildInputs ++ [ pkg-config ];
+    nativeBuildInputs = prev.nativeBuildInputs ++ [ pkg-config ];
     patches = (prev.patches or [ ]) ++ [
       (fetchpatch {
         name = "stormlib-optimizations.patch";

